recipes/patchelf/all/conanfile.py

```diff
@@ -1,57 +1,79 @@
+from conan.tools.gnu import Autotools, AutotoolsDeps, AutotoolsToolchain
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+
 import os
 
-required_conan_version = ">=1.33.0"
+required_conan_version = ">=1.54.0"
 
 class PatchElfConan(ConanFile):
     name = "patchelf"
+    package_type = "application"
     description = "A small utility to modify the dynamic linker and RPATH of ELF executables"
     topics = ("conan", "elf", "linker", "interpreter", "RPATH", "binaries")
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/NixOS/patchelf"
     license = "GPL-3.0-or-later"
     settings = "os", "arch", "compiler", "build_type"
 
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def build_requirements(self):
-        self.build_requires("libtool/2.4.6")
+        self.tool_requires("libtool/2.4.6")
 
     def validate(self):
-        if not tools.is_apple_os(self.settings.os) and self.settings.os not in ("FreeBSD", "Linux"):
+        if not is_apple_os(self) and self.settings.os not in ("FreeBSD", "Linux"):
             raise ConanInvalidConfiguration("PatchELF is only available for GNU-like operating systems (e.g. Linux)")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version][0], strip_root=True)
+        # The library sources do not contain a license file, so download it from elsewhere
+        download(self, filename="LICENSE", **self.conan_data["sources"][self.version][1])
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
 
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self)
-        self._autotools.configure(configure_dir=self._source_subfolder)
-        return self._autotools
+        tc = AutotoolsToolchain(self)
+        tc.generate()
+
+        tc = AutotoolsDeps(self)
+        tc.generate()
 
     def build(self):
-        with tools.chdir(self._source_subfolder):
-            self.run("{} -fiv --warnings=all".format(tools.get_env("AUTORECONF")), run_environment=True)
-        autotools = self._configure_autotools()
+        autotools = Autotools(self)
+        autotools.autoreconf()
+        autotools.configure()
         autotools.make()
 
     def package(self):
-        self.copy(pattern="COPYING", src=self._source_subfolder, dst="licenses")
-        autotools = self._configure_autotools()
+        copy(self, pattern="LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
```

Comment:
As per previous discussion
```suggestion
        copy(self, pattern="COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
```